router/core/request_compression_test.go

```diff
@@ -0,0 +1,207 @@
+package core
+
+import (
+	"compress/flate"
+	"compress/gzip"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	br "github.com/andybalholm/brotli"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+var responses = map[string]string{
+	"text/html":                "<html><body>Hello World !</body></html>",
+	"text/css":                 "body { color: black; }",
+	"text/plain":               "Hello World !",
+	"text/javascript":          "console.log('Hello World !');",
+	"application/javascript":   "console.log('Hello World !');",
+	"application/x-javascript": "console.log('Hello World !');",
+	"application/json":         `{"message": "Hello World !"}`,
+	"application/atom+xml":     `<feed><title>Hello World !</title></feed>`,
+	"application/rss+xml":      `<rss><channel><title>Hello World !</title></channel></rss>`,
+	"image/svg+xml":            `<svg><text>Hello World !</text></svg>`,
+	"application/graphql":      `query { message }`,
+}
+
+func TestCompression(t *testing.T) {
```

Comment:
What is the value of unit testing if dependencies are working? Please put the most important tests into integration tests.

Reply:
I also feel that unit test for this is unnecessary. Only integration test is necessary. I will remove the unit test.